lib/beaker/host/unix/pkg.rb

```diff
@@ -33,6 +33,9 @@ def check_for_package(name, opts = {})
           # time. It's just to make sure that we only do the key import once, &
           # isn't for setting or use outside of beaker.
           execute('rpmkeys --import http://nightlies.puppetlabs.com/07BB6C57', opts)
+
+          # We also need the real key in case users are pulling from the stable packages
+          execute('rpmkeys --import https://yum.puppetlabs.com/RPM-GPG-KEY-puppet', opts)
```

Comment:
I've recently started work to try to fix the sles repos

I've added https://github.com/puppetlabs/beaker/pull/1449, which alone isn't going to fix this. But I'm also working on getting our repo automation updated so it signs the yum repos, and provides the gpg key in the repo. With that, zypper will automatically import the key in the repo if directed to do so. This opens us up to transients for network blips, but it's certainly better than what we have. However, this is also a perfectly reasonable solution to fix these failures. I'm wondering if this key is available under `/etc/pki/rpm-gpg/RPM-GPG-KEY-puppet-PC1`. It would only be dropped there by the puppetlabs-release package. I'm not sure if that's getting installed or not